### PR TITLE
Handle ui test errors in Jenkins Pipeline

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Fix jenkins testing stage to not ignore test result. This led to different results between local builds and Jenkins builds which confused the developer and could hide implementation errors. 

**Notice**: This is build is currently broken on purpose to show the actual pipeline changes. This pull requests itself doesn't break the build and should only affect the error handling inside the pipeline.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [ ] New code is covered with unit tests
- [ ] New/updated ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog`
- [ ] Feature has been tested with different permissions
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-ui/build.gradle [hunk 1]
```diff
@@ -84,11 +84,10 @@ task test(type: YarnTask) {
     include 'ui-*/**/*.ts'
     include 'ui-*/**/*.tsx'
   })
-  .withPathSensitivity(PathSensitivity.RELATIVE)
+    .withPathSensitivity(PathSensitivity.RELATIVE)
   outputs.dir('build/jest-reports')
 
   if (project.isCI) {
-    ignoreExitValue = project.isCI
```
Comment (eheimbuch) [inline scm-ui/build.gradle:91]: This is the actual change in the gradle task. Exit codes are not longer ignored by Jenkins but handled correctly in the build.

File: Jenkinsfile [hunk 2]
```diff
@@ -61,10 +61,12 @@ pipeline {
 
     stage('Check') {
       steps {
-        withCheckEnvironment {
-          gradle 'check'
+        catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
```
Comment (sdorra) [inline Jenkinsfile:64]: Why do we set the build result to unstable, but the stage result to failure?

Comment (sdorra) [inline Jenkinsfile:64]: Why do we set the build to unstable, but the stage to failure? Is the rest of the stage still executed, after setting a stage result?

Comment (sdorra) [inline Jenkinsfile:64]: Why do we set the build to unstable, but the stage to failure? Is the rest of the stage still executed, after setting a stage result?

File: Jenkinsfile [hunk 3]
```diff
@@ -61,10 +61,12 @@ pipeline {
...
+            withCheckEnvironment {
+              gradle 'check'
+            }
+            junit allowEmptyResults: true, testResults: '**/build/test-results/test/TEST-*.xml,**/build/test-results/tests/test/TEST-*.xml,**/build/jest-reports/TEST-*.xml'
```
Comment (sdorra) [inline Jenkinsfile:68]: I think we should collect the test results after the catch. This allows us to see the test results of the modules, even if a module fails.

Comment (sdorra) [inline Jenkinsfile:68]: I think we should collect the test results after the catch. This allows us to see the test results of the modules, even if a module fails.

**Review by sdorra — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (eheimbuch) [reply Jenkinsfile:64]: Yes, this is the topic this pr is all about.

**Pull request merged**